### ISSUE TITLE
NewType and TypeVar incorrect usage not detected by pylance

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pyright#3950]
## Environment data

-   Language Server version: Pylance insiders v2022.9.21
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.3

## Code Snippet

```python
from typing import NewType, Type, TypeVar

a = 3
b = int
c = type(a)
d = TypeVar('d', bound=int)
e = NewType('e', int)


w = e('0x00f0', base=16)
x = b('0x0034', base=16)    
y = c('0x00FF', base=16)
z = d('0x00fe', base=16)

print(x)
print(y)
print(z)
print(w)
```

## Repro Steps

Run mpy against this file. This is the output I get

```
PS C:\Users\aku91> mypy test.py
test.py:10: error: Unexpected keyword argument "base" for "e"
test.py:10: error: Argument 1 to "e" has incompatible type "str"; expected "int"
test.py:13: error: "object" not callable
Found 3 errors in 1 file (checked 1 source file)
PS C:\Users\aku91> 
```

## Expected behavior

Pylance should detect the two problems:
- NewType is used incorrectly
- TypeVar type is not constructable

## Logs

[pylancelog.log](https://github.com/microsoft/pylance-release/files/9577216/pylancelog.log)

**Comment by erictraut** [comment]
This looks to be working correctly. If you enable type checking, you should see the following errors, which are the same errors reported by mypy.

![Screen Shot 2022-09-15 at 10 18 36 AM](https://user-images.githubusercontent.com/7040122/190469333-f0e10162-8c3a-4398-bd41-0263d1d9c8c1.png)

**Comment by rchiodo** [comment]
In case somebody else hits this, the setting is here:

```json
    "python.analysis.typeCheckingMode": "basic"

```
